sdcm/nemesis.py

```diff
@@ -3820,6 +3821,122 @@ def _shrink_cluster(self, rack=None):
+                            'aws_use_ec2_credentials': True,
+                        }
+                        is_restart_needed = True
+                if is_restart_needed:
+                    node.configure_kms()
+                    node.restart_scylla()
+
+        # Create table with encryption
+        keyspace_name, table_name = self.cluster.get_test_keyspaces()[0], 'tmp_encrypted_table'
+        self.log.info("Create '%s.%s' table with server encryption", keyspace_name, table_name)
+        with self.cluster.cql_connection_patient(self.target_node, keyspace=keyspace_name) as session:
+            # NOTE: scylla-bench expects following table structure:
+            #       (pk bigint, ck bigint, v blob, PRIMARY KEY(pk, ck)) WITH compression = { }
+            create_table_query_cmd = (
+                f"CREATE TABLE IF NOT EXISTS {table_name}"
+                " (pk bigint, ck bigint, v blob, PRIMARY KEY (pk, ck))"
+                " WITH compression = { } AND read_repair_chance=0.0"
+                f" AND compaction = {{ 'class' : '{self.cluster.params.get('compaction_strategy')}' }}"
+                f" AND scylla_encryption_options = {scylla_encryption_options};")
+            session.execute(create_table_query_cmd)
+
+        def upgrade_sstables(nodes):
+            for node in nodes:
+                self.log.info("Upgradesstables on the '%s' node for the new encrypted table", node.name)
+                node.remoter.run(f'nodetool upgradesstables -a -- {keyspace_name} {table_name}', verbose=True)
+
+        try:
+            for _ in range(2 if (aws_kms and kms_key_alias_name) else 1):
+                # Write data
+                write_cmd = (
+                    "scylla-bench -mode=write -workload=sequential -consistency-level=all -replication-factor=3"
+                    " -partition-count=50 -clustering-row-count=100 -clustering-row-size=uniform:50..150"
+                    f" -keyspace {keyspace_name} -table {table_name} -timeout=120s")
+                write_thread = self.tester.run_stress_thread(stress_cmd=write_cmd, stop_test_on_failure=False)
+                self.tester.verify_stress_thread(write_thread)
+                upgrade_sstables(self.cluster.nodes)
+
+                # Read data
+                read_cmd = (
+                    "scylla-bench -mode=read -workload=sequential -consistency-level=all"
+                    " -partition-count=50 -clustering-row-count=100 -timeout=60s -iterations=1 -validate-data"
+                    " -concurrency=10 -connection-count=10 -rows-per-request=10")
+                read_thread = self.tester.run_stress_thread(stress_cmd=read_cmd, stop_test_on_failure=False)
+                self.tester.verify_stress_thread(read_thread)
+
+                # Rotate KMS key if applicable,
+                # do it 2nd time too for 2nd+ nemesis runs which will fail to create already existing alias
+                if aws_kms and kms_key_alias_name:
```

Comment:
I'm sure it's a good idea to mix the enable/disable nemesis with the key rotation 
Enable/Disable can be a more general for EaR, while this rotating is KMS only.

I know the flows might be close, but I would expect two different Nemesis

Reply:
I'm **not** eating today... maybe that's the reason :)

anyhow I still would like to see Enable/Disable without rotation case happening, and like this rotation is always gonna happen on the enable disable for KMS. I think it's two distinct cases and we should try keeping them as such.

Also I would like to see a rotation on the main load table, while it's written into it, on a case running KMS to being with.

Reply:
For now I will add a boolean flag keeping the logic in place for case we want to use rotation in the nemesis.